src/subject/subject.service.ts

```diff
@@ -158,15 +206,29 @@ export class SubjectService {
               where: { is_active: true },
             },
           },
-          where: {
-            id_status: {
-              in: [MonitorStatus.AVAILABLE, MonitorStatus.APPROVED],
-            },
-          },
+          where: monitorStatusFilter,
         },
       },
       orderBy: { name: 'asc' },
     });
+
+    if (data.length === 0 && teacherId) {
+      throw new NotFoundException('Professor não encontrado.');
+    }
```

Comment:
Essa checagem tem que ser a primeira coisa feita pelo método se o teacherId for passado

Reply:
Você deve procurar pro um registro na tabela `teacher` com o ID passado pra saber se ele existe ou não. A forma como você verificou não é muito segura e pode se tornar um bug